backend/cmd/headlamp.go

```diff
@@ -110,13 +111,35 @@ func copyReplace(src string, dst string,
 	}
 }
 
+func strInFile(str, filepath string) bool {
+	contents, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		panic(err)
+	}
+
+	isThere, err := regexp.Match(str, contents)
+	if err != nil {
+		panic(err)
+	}
+
+	return isThere
+}
+
 // make sure the base-url is updated in the index.html file.
 func baseURLReplace(staticDir string, baseURL string) {
 	indexBaseURL := path.Join(staticDir, "index.baseUrl.html")
 	index := path.Join(staticDir, "index.html")
 
+	if baseURL == "" && strInFile("headlampBaseUrl=\".\"", index) {
+		// The index.html does not need resetting from a different baseURL.
+		return
+	}
+
```

Comment:
IIUC, the `index.baseUrl.html` will always have the original value of `index.html`, right? Also, the `index.baseUrl.html` should only exist if we override the baseUrl (at least with these new changes).

So an extra check which prevents the reading would be to check if the indexBaseUrl file exists and early return if it doesn't.

Reply:
Good idea. I added the extra check to see if the file is missing first to avoid the read.